docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -350,8 +350,41 @@ Excel.run(function (context) {
 
 You can also find the cell responsible for spilling into a given cell by using the [Range.getSpillParent](/javascript/api/excel/excel.range#getspillparent--) method. Note that `getSpillParent` only works when the range object is a single cell. Calling `getSpillParent` on a range with multiple cells will result in an error being thrown (or a null range being returned for `Range.getSpillParentOrNullObject`).
 
+## Trace formula precedents
+
+An Excel formula often refers to other cells or groups of cells. These cells or groups of cells are known as "precedents" to the formula. To learn more about Excel features related to relationships between cells, see the [Display the relationships between formulas and cells](https://support.microsoft.com/office/display-the-relationships-between-formulas-and-cells-a59bef2b-3701-46bf-8ff1-d3518771d507) article. 
+
+With [Range.getDirectPrecedents](/javascript/api/excel/excel.range#getdirectprecedents--), your add-in can locate the direct precedents to a formula. The `getDirectPrecedents` method can trace precedents to a formula within a single worksheet and across worksheets within a single workbook. This method can't trace precedent cells across workbooks. 
```

Comment:
Can we describe the returned `WorkbookRangeAreas` object here? It's a complex data structure that could use some object model context. Maybe something like "`Range.getDirectPrecedents` returns a `WorkbookRangeAreas` object. This object contains the address of all the precedents in the workbook. It has a separate `RangeAreas` object for each worksheet containing at least one precedent for the formula. See "Work with multiple ranges simultaneously in Excel add-ins" for more information on working with the `RangeAreas` object."

Saying all that, it might also be worth adding a section about WorkbookRangeAreas to the multiple ranges article. I'll leave that to your discretion.

Reply:
Thanks for the explanation! I will look into adding a new section to the multiple ranges article.